### ISSUE TITLE
Lost GPU acceleration after last Chrome OS updates

### ISSUE THREAD

**Issue by javlik** [dnschneid/crouton#4271]
```
name: xenial
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/xenial...
crouton: version 1-20200428114634~master:67dc8d77
release: xenial
architecture: amd64
xmethod: xorg
targets: xfce,xorg
host: version 12871.76.0 (Official Build) stable-channel edgar 
kernel: Linux localhost 4.19.98-08081-g68e60340b630 #1 SMP PREEMPT Fri Apr 10 00:52:47 PDT 2020 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...
```

I've been using crouton for several years without any problem - actually, I love it.
Now, after the last Chrome OS update (whose version was 81.0.4044.103) the 3D acceleration stopped working completely.

I tried several powerwashes, new installs ... nothing helped.

**Comment by jonasthenoob** [comment]
Having the exact same problem with xenial.
However i did try focal* release which is unsupported but there was not lag or nothing and it was completely usable so i think this is a problem with xenial

**Comment by javlik** [comment]
Thanks a lot. The focal* seems to be working fine (despite being unsupported), but there is another issue with this target: Audio does not work at all here.
So, is this a known issue? Is there a fix? Or, should I create a new issue?

**Comment by jonasthenoob** [comment]
i think you should but yeah focal* seems to be working fine however the output and input is not working.
I will reply asap, right now i am going to try out debian and see if that works

**Comment by jonasthenoob** [comment]
i found a fix to this all audio working as well.
So download xenial however u want with any of your favorite targets and then when thats 
installed do not use the sudo start(targetname) just do sudo enter-chroot and then do sudo do-release-upgrade and it should upgrade to bionic 18.04. After the upgrade is done update the files by running sudo apt-get upgrade and then when that upgrades there should be no lag! 
Hope this works for you too.

**Comment by jvschiavo** [comment]
Wow! So that's probably what happened to me as well! I can't play any Steam game anymore without terrible FPS. :/
I miss my Spelunky, The Binding of Isaac Rebirth, Magicite... or really any other game. Everything lags terribly and is unplayable, and I didn't know the cause. I though it was mesa driver that needed to be updated, but the kisak ppa is no longer compatible with xenial, so I was trying to update to bionic anyway. Actually I'm doing that now.

Btw, I have a Samsung Chromebook 2 on 81.0.4044.127.

@DennisLfromGA I think this shouldn't be marked with a question label, this is definitely a bug.

> The focal* seems to be working fine (despite being unsupported), but there is another issue with this target: Audio does not work at all here. [...] should I create a new issue?

@javlik I think you should create a new issue about this audio problem.

**Comment by DennisLfromGA** [comment]
@jvschiavo,

I marked it as a question to garner the attention of the crouton devs.

It may indeed be a bug but categorizing it as a 'bug', an 'upstream bug' or a 'crosbug' is not clear at this point - to me at least.

However, I've removed the 'question' label as you requested.

**Comment by jvschiavo** [comment]
I'm doing some tests. I managed to upgrade from 16.04 to 18.04 with no big problems, and then to 20.04 with a lot of headache (dependency hell + lots of bugs), but my games are still lagging... at least Spelunky is, and I was playing fine with good fps before v81.

@javlik could you lend me some help on where do I know for sure if I have 3D acceleration enabled or not? I have no idea where do I find that.

**Comment by jonasthenoob** [comment]
i would stick with 18.04 it's more stable

**Comment by javlik** [comment]
@jonasthenoob Thank you again. I tried to fix the sound issues in the *focal and "somehow" managed it. I will check if I am able to repeat all the steps and if so, I will put them here. I am also trying the "xenial -> bionic" way but if there is no other problem with *focal, I will probably stick with this version.
@jvschiavo Usually, I just install the mesa-utils package and then simply check out how smooth is animation in e.g. glxgears. But, since I was dealing also with the sound issue, I installed the great game neverputt from ubuntu repositories to check both, graphics and sound. In focal*, after the mentioned fix, both problems are now away and everything seems to be working. If I encounter another problem I will raise a new issue.
p.s.: Sorry for my english - It's not my mother tongue...

**Comment by jvschiavo** [comment]
> but my games are still lagging... at least Spelunky is

I rectify my last comment, Spelunky is running fine again, with good FPS, I just had to change the Proton version I was using. :)

I may or may not check again in 18.04, but it seems that I got 3D acceleration working in 20.04.

This is totally a bug with xenial and (probably) Chrome OS v81, I was playing games just fine before that version IIRC. Sadly the crouton update script (`sudo crouton -u -n xenial`) cannot fix the problem.

@jonasthenoob thanks for the suggestion. I made a backup when I upgraded to 18.04 bionic before upgrading again, so I will just restore. I don't actually intend to use 20.04 for now (or even 18.04), I'm just hyped to get it on the crouton support list ASAP so I can use it with confidence.
